### DISCUSSION TITLE
math in markdown via KaTeX/MathJax

### DISCUSSION THREAD

**Discussion by nschloe** [community/community#4991] [Other Feature Feedback, Questions, & Ideas] [answered]
GitLab has several useful extensions in their Markdown support. Most important for the scientific community arguably is [math support](https://docs.gitlab.com/ee/user/markdown.html#math). Would love to see this in GitHub, too.

_Edit:_ I added this in the browser extension [xhub](https://github.com/nschloe/xhub), but native support would still make a big difference.

```math
f(a) = \frac{1}{2\pi i} \oint _{\gamma}\frac{f(z)}{z-a} dz.
```
````markdown
```math
f(a) = \frac{1}{2\pi i} \oint _{\gamma}\frac{f(z)}{z-a} dz.
```
````

**Comment by davidstaheli** [comment, marked as answer]
Thank you for posting this, @nschloe. We're actively working on it.

**Reply by pattacini** [reply]
MathJax support is now available 🎉 
https://github.blog/changelog/2022-05-19-render-mathematical-expressions-in-markdown/

**Reply by nschloe** [reply]
It's pretty buggy though. :frowning_face:

**Reply by pattacini** [reply]
Yes, indeed. Just an example: #16944.
I'm still using your nice [`xhub`](https://github.com/nschloe/xhub)!

**Comment by Lemmingh** [comment]
@davidstaheli

The current processing model is **unnecessarily complicated**, forcing users to understand how Markdown (`cmark-gfm`), [GitHub's pipeline](<https://github.blog/2012-11-27-html-pipeline-chainable-content-filters/>), HTML, DOM, and MathJax work.

For example,

This is recognized as math:

```text
&dollar;\frac{f}{a}&dollar;
```

But this is recognized as link:

```text
$[(a+b)!](c+d)$
```

![2022-05-20_GitHub math WTF](https://user-images.githubusercontent.com/43396014/169503649-de0017ab-822e-490d-8d15-5a90eb38a5eb.png)

---

Why not to follow GitLab's solution?

GitLab protects LaTeX expressions with code spans and code blocks, which avoids all the [hells](<https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions#writing-dollar-signs-in-line-with-and-within-mathematical-expressions>)!

**Reply by nschloe** [reply]
I agree with this. There is something substantially wrong with how the Markdown parser and the math parser are competing. This results in various bugs and ambiguous behavior as outlined here. I've written up a blog post about it, too:

https://nschloe.github.io/2022/05/20/math-on-github.html
